rust/scx_utils/src/topology.rs

```diff
@@ -460,13 +494,41 @@ fn create_insert_cpu(
     let base_freq = read_file_usize(&freq_path.join("base_frequency")).unwrap_or(max_freq);
     let trans_lat_ns = read_file_usize(&freq_path.join("cpuinfo_transition_latency")).unwrap_or(0);
 
-    let llc = node.llcs.entry(llc_id).or_insert(Arc::new(Llc {
-        id: llc_id,
+    let multi_numa = topo_ctx.node_llc_kernel_ids.len() > 0;
+    let node_llc_kernel_ids: &mut BTreeMap<usize, usize> =
+        if topo_ctx.node_llc_kernel_ids.contains_key(&node.id) {
+            topo_ctx
+                .node_llc_kernel_ids
+                .get_mut(&node.id)
+                .context("impossible")?
+        } else {
+            let core_kernel_ids = BTreeMap::new();
+            topo_ctx
+                .node_llc_kernel_ids
+                .insert(node.id, core_kernel_ids.clone());
+            topo_ctx
+                .node_llc_kernel_ids
+                .get_mut(&node.id)
+                .context("impossible")?
+        };
+    if !node_llc_kernel_ids.contains_key(&llc_kernel_id) {
+        if node_llc_kernel_ids.len() > 0 || multi_numa {
+            topo_ctx.llc_id += 1;
+        }
+        node_llc_kernel_ids.insert(llc_kernel_id, topo_ctx.llc_id);
+    }
```

Comment:
Wouldn't it be easier to make it `BTreeMap<(node_id, llc_kernel_id), llc_id>`? Then there is only one map and the new `llc_id` would just be `map.len()`.

Reply:
Yeah, that's more reasonable 👍